### ISSUE TITLE
SMTP bcc doesn't seem to be working

### ISSUE THREAD

**Issue by ptdev** [bcosca/fatfree-core#130]
Hi there, I've just tried to use f3 to send some emails with bcc and it appears as the bcc recipient isn't receiving the emails.
I saw that one of the last commits on the smtp.php file was regarding the bcc and cc headers being ignored so I updated to the latest version but still no go.

my test:
```
$mail = new \SMTP();
$mail->set('From', 'system <test@example.com>');
$mail->set('Subject', 'test subject');
$mail->set('To', 'some name <somebody@example.com>');
$mail->set('Bcc', 'somebody else <thebcc@example.com>');
$mail->send('testing');
```
Can somebody please run a simple test on their side and confirm if this is also happening for them?
Thanks.

**Comment by KOTRET** [comment]
mhm...
```php
$email='';
if (in_array($key,['From','To','Cc','Bcc']))
	foreach (explode(',',$val) as $raw)
		if (!preg_match('/[<>]/',$raw))
			$email.=($email?',':'').'<'.$raw.'>';
$val=$email;
```

the code reads like this for me:
if each email-address `$raw` (exploded by `,`) in from/to/cc/bcc does not contain `<` or `>` then add `<$raw>` to the `$email`. I miss the part when the address matches.

This should fix it (untested though, just written here)

```php
$email='';
if (in_array($key,['From','To','Cc','Bcc']))
	foreach (explode(',',$val) as $raw)
		if (!preg_match('/[<>]/',$raw))
			$raw='<'.$raw.'>';
		$email.=($email?',':'').$raw;
$val=$email;
```

beside that: an email-address may look like this: `"Interessierts, Wayne"<wayne.interessierts@example.com>`, so splitting by comma is error-prone.

**Comment by ptdev** [comment]
So, a better solution to also avoid the issue you described could maybe be something like:

When only email is available:
```
$mail->set('From/To/Cc/Bcc', 'somebody@example.com');
```

When email and name are available:
```
$mail->set('From/To/Cc/Bcc', ['name' => 'somebody@example.com']);
```

And then detect if the parameter is an array or a string and build the headers accordingly.
Thoughts?

**Comment by KOTRET** [comment]
you can have multiple recipients. As we are working on the header directly it is a bad idea to replace this with some sort of array.

**Comment by ptdev** [comment]
Ah yes, didn't remember that as I rarely use more than 1 address on each header. One could still call set() once for each recipient and have it build the correct headers internally. But ok I agree that it would change the way the setter is used and probably break backwards compatibility so maybe not such a good idea.

Anyway, I tried your change but still got the same issue so I've reverted the file back to "3.5.1-Release" and that finally fixed it.
